### ISSUE TITLE
Question for the compute_z/compute_v while applying margin_loss and entropy_loss

### ISSUE THREAD

**Issue by JasonH0810** [Vaidehi99/InfoDeletionAttacks#9]
Hi, 

Thank you for your great work! I just got a question while reproducing your work:

Question: Why is nll_loss excluded when fact_erasure is used with margin_loss or entropy_loss?

In the snippet below, when args.fact_erasure or args.margin_loss (or args.entropy_loss) are enabled, the total loss omits nll_loss. In other branches (e.g., without margin/entropy, or outside fact_erasure), nll_loss (or a probability derived from it) is included. Is this intentional or an oversight?

if args.fact_erasure:
    if args.margin_loss:
        loss = kl_loss + weight_decay + margin_loss_top + margin_loss_bottom
        print( ... uses torch.exp(-nll_loss_each).mean() ... )
    elif args.entropy_loss:
        loss = -entropy + kl_loss + weight_decay
        print( ... uses torch.exp(-nll_loss_each).mean() ... )
    else:
        pred_prob = torch.exp(-nll_loss)
        loss = pred_prob + kl_loss + weight_decay
else:
    loss = nll_loss + kl_loss + weight_decay
    print( ... uses torch.exp(-nll_loss_each).mean() ... )

Is the intention that, under fact_erasure, margin_loss or -entropy replaces nll_loss as the primary objective (e.g., to avoid double-counting or gradient conflicts)? If yes, could this be documented explicitly?

If not, should the total loss be something like:
    - loss = nll_loss + kl_loss + weight_decay + margin_loss_top + margin_loss_bottom, or
    - loss = nll_loss - alpha * entropy + kl_loss + weight_decay (with tunable weights)?

Expected outcome:
- Clarify whether this is intentional;
- If not intentional, consider adding nll_loss back (with a default weight);
- Or document why excluding nll_loss is preferable here.

Thanks a lot for the great work and for open-sourcing it! 🙏
